### PR TITLE
RELATED: RAIL-3383 - Add dashboard config & code to load it

### PR DESCRIPTION
**Pull request by lupko**
New 'config' part of state. LoadDashboard expanded to load the configuration if the dashboard does not get it from props.

Moved code from KD.

Established model/_staging directory. Code moved from KD that should belong to other package (filters in this case) is staged here during the initial phases.

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     },
     "scripts": {
         "clean": "rm -rf ci dist esm coverage *.log styles/css && jest --clearCache",
-        "api-extractor": "mkdir -p api && [ -z \"${CI}\" ] && (api-extractor run -l) || (api-extractor run)",
+        "api-extractor": "mkdir -p api && [ -z \"${CI}\" ] && (api-extractor run -l; exit 0) || (api-extractor run)",
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/package.json:39]: What is the reason for this change?

File: libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -41,8 +42,17 @@ export interface IDashboardProps {
      */
     dashboardRef?: ObjRef;
 
+    /**
+     * Configuration that modifies to dashboard features, capabilities and behavior.
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx:46]: Nitpick: strange wording: modifies to dashboard

File: libs/sdk-ui-dashboard/src/model/index.ts [hunk 3]
```diff
@@ -37,3 +46,7 @@ export {
     DashboardCommandType,
 } from "./commands/dashboard";
 export { DashboardContext } from "./types/commonTypes";
+
+export { DashboardConfig } from "./types/commonTypes";
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/index.ts:50]: Nitpick: duplicate import (the previous one has the same source file.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export function* loadDashboardCommandHandler(ctx: DashboardContext, cmd: LoadDashboard) {
+    // eslint-disable-next-line no-console
+    console.debug("handling load dashboard", cmd, "in context", ctx);
+
+    try {
+        yield put(loadingActions.setLoadingStart());
+
+        // TODO: how to type this properly?
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts:28]: PromiseFnReturnType ?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts (hunk 4 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts:28]: the `all` is messing me up here, for normal `call` i used the PromiseFnReturnType. but here i'm at loss how to use it

File: libs/sdk-ui-dashboard/package.json (hunk 1 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/package.json:39]: if api extractor run fails, it will otherwise run second time (due to exit 1 and ||)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3851/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5481/ : FAILURE in 5m 57s

File: libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx (hunk 2 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx]: fixed

File: libs/sdk-ui-dashboard/src/model/index.ts (hunk 3 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/index.ts]: fixed

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3852/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5482/ : FAILURE in 5m 45s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts (hunk 4 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/loadDashboard/handler.ts]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3854/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5484/ : SUCCESS in 15m 00s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1126/ : FAILURE in 2m 45s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1396/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/132/ : FAILURE in 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5485/ : SUCCESS in 15m 00s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1127/ : FAILURE in 2m 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1397/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/133/ : FAILURE in 26s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5486/ : SUCCESS in 15m 06s

**Comment by lupko** [comment]
extended check sonar

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3856/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/135/ : FAILURE in 26s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3855/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5488/ : SUCCESS in 14m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3862/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5492/ : SUCCESS in 15m 07s

**Comment by lupko** [comment]
extended check sonar

**Comment by lupko** [comment]
extended check sonar

**Comment by gdgate** [comment]
Build in pipeline check-extended-sonar aborted.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3865/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/139/ : SUCCESS in 8m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3864/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5494/ : SUCCESS in 15m 17s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1130/ : FAILURE in 2m 36s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1400/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/141/ : SUCCESS in 9m 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5497/ : SUCCESS in 18m 01s
